xdsl/dialects/fsm.py

```diff
@@ -0,0 +1,377 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    ContainerOf,
+    DictionaryAttr,
+    FlatSymbolRefAttr,
+    FunctionType,
+    IndexType,
+    IntegerType,
+    StringAttr,
+)
+from xdsl.ir import (
+    Attribute,
+    Block,
+    Operation,
+    Region,
+    SSAValue,
+)
+from xdsl.ir.core import Dialect
+from xdsl.irdl import (
+    AnyAttr,
+    AnyOf,
+    IRDLOperation,
+    attr_def,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    opt_operand_def,
+    region_def,
+    var_operand_def,
+    var_result_def,
+)
+from xdsl.traits import (
+    HasParent,
+    IsTerminator,
+    NoTerminator,
+    SymbolOpInterface,
+    SymbolTable,
+)
+from xdsl.utils.exceptions import VerifyException
+
+signlessIntegerLike = ContainerOf(AnyOf([IntegerType, IndexType]))
+
```

Comment:
Maybe it makes sense make clear within this file that you are mirroring the CIRCT FSM ops, so people know where to look for additional documentation, since this is not part of MLIR core?